src/core/main.cpp

```diff
@@ -528,6 +528,57 @@ int main( int argc, char * * argv )
 		}
 	}
 
+	// Test file argument before continuing
+	QFileInfo fileInfoLoad( fileToLoad );
+	QFileInfo fileInfoImport( fileToImport );
+	if( !fileToLoad.isEmpty() )
+	{
+		if( !fileInfoLoad.exists() )
+		{
+			printf( "The file %s does not exist!\n",
+					fileToLoad.toStdString().c_str() );
+			exit( 1 );
+		}
+		else if( fileInfoLoad.isDir() )
+		{
+			printf( "%s is a directory!\n",
+					fileToLoad.toStdString().c_str() );
+			exit( 1 );
+		}
+		else if( !( fileInfoLoad.suffix().toLower() == "mmp" ||
+					fileInfoLoad.suffix().toLower() == "mmpz" ) )
+		{
+			printf( "%s is not an LMMS project file!\n",
+					fileToLoad.toStdString().c_str() );
+			exit( 1 );
+		}
+	}
+	else if( !fileToImport.isEmpty() )
+	{
+		if( !fileInfoImport.exists() )
+		{
+			printf( "The file %s does not exist!\n",
+					fileToImport.toStdString().c_str() );
+			exit( 1 );
+		}
+		else if( fileInfoImport.isDir() )
+		{
+			printf( "%s is a directory!\n",
+					fileToImport.toStdString().c_str() );
+			exit( 1 );
+		}
+		else if( !(
+					fileInfoImport.suffix().toLower() == "mid"	||
+					fileInfoImport.suffix().toLower() == "midi"	||
+					fileInfoImport.suffix().toLower() == "rmi"	||
+					fileInfoImport.suffix().toLower() == "flp"	||
+					fileInfoImport.suffix().toLower() == "h2song" ) )
```

Comment:
Hmm.... Every time I see a laundry list of file extensions I get worried that we're making more work for ourselves down the road.  😄  Perhaps related to this PR, perhaps not.... can we start managing our extension in a centralized place?  We should be able to use `#ifdef`'s to enable/disable them globally based on what libraries/plugins cmake finds.  We already do this in several other places already.  Food for thought.